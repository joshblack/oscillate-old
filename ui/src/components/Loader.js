import React from 'react';
import cx from 'classnames';

const isIe = window.ActiveXObject || 'ActiveXObject' in window;

const Loader = ({ active }) => {
  const className = cx(
    'loader',
    {
      'loader--ie': isIe,
      'loader--stop': !active,
      'loader--stop--ie': !active && isIe,
    },
  );

  return (
    <div className={className}>
      <svg className="loader__svg" viewBox="-75 -75 150 150">
        <circle cx="0" cy="0" r="37.5" />
      </svg>
    </div>
  );
};

Loader.propTypes = {
  active: React.PropTypes.bool.isRequired,
};

export default Loader;
