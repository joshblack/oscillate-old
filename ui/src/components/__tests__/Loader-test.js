import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../Loader';

describe('Loader component', () => {
  it('should render', () => {
    const active = renderer.create(
      <Loader active={true} />
    ).toJSON();
    expect(active).toMatchSnapshot();

    const exit = renderer.create(
      <Loader active={false} />
    ).toJSON();
    expect(exit).toMatchSnapshot();
  });
});
