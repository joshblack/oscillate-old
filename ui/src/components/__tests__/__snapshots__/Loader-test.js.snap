exports[`Loader component should render 1`] = `
<div
  className="loader">
  <svg
    className="loader__svg"
    viewBox="-75 -75 150 150">
    <circle
      cx="0"
      cy="0"
      r="37.5" />
  </svg>
</div>
`;

exports[`Loader component should render 2`] = `
<div
  className="loader loader--stop">
  <svg
    className="loader__svg"
    viewBox="-75 -75 150 150">
    <circle
      cx="0"
      cy="0"
      r="37.5" />
  </svg>
</div>
`;
